jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitCredentialProcessProvider.kt

```diff
@@ -0,0 +1,101 @@
+    }
+    private val cmd by lazy {
+        if (SystemInfo.isWindows) {
+            GeneralCommandLine("cmd", "/C", command)
+        } else {
+            GeneralCommandLine("sh", "-c", command)
+        }
+    }
+    private val processCredentialCache = CachedSupplier.builder { refresh() }.build()
+
+    override fun resolveCredentials(): AwsCredentials = processCredentialCache.get()
+
+    private fun refresh(): RefreshResult<AwsCredentials> {
+        val timeout = Registry.intValue("aws.credentialProcess.timeout", DEFAULT_TIMEOUT)
+        val output = ExecUtil.execAndGetOutput(cmd, timeout)
+
+        if (output.isTimeout) {
+            handleException(message("credentials.profile.credential_process.timeout_exception_prefix", entrypoint), output)
+        }
+
+        if (output.exitCode != 0) {
+            handleException(message("credentials.profile.credential_process.execution_exception_prefix", entrypoint), output)
+        }
+
+        val result = try {
+            parser.parse(output.stdout)
+        } catch (e: Exception) {
+            handleException(message("credentials.profile.credential_process.parse_exception_prefix"), output, e)
+        }
+        val credentials = when (val token = result.sessionToken) {
+            null -> AwsBasicCredentials.create(result.accessKeyId, result.secretAccessKey)
+            else -> AwsSessionCredentials.create(result.accessKeyId, result.secretAccessKey, token)
+        }
+        return RefreshResult.builder(credentials).staleTime(result.expiration ?: Instant.MAX).build()
+    }
+
+    private fun handleException(msgPrefix: String, process: ProcessOutput, exception: Exception? = null): Nothing {
+        val errorOutput = process.stderr.takeIf { it.isNotBlank() }
+        val details = errorOutput ?: exception?.message?.takeIf { it.isNotBlank() } ?: exception?.cause?.message?.takeIf { it.isNotBlank() }
+        val msg = "$msgPrefix${details?.let { ": $it" } ?: ""}"
+        LOG.debug { process.stdout }
+        LOG.error(exception) { msg }
+        throw RuntimeException(msg, exception)
+    }
+
+    internal companion object {
+        private val LOG = getLogger<ToolkitCredentialProcessProvider>()
+        private const val DEFAULT_TIMEOUT = 30000
+    }
+}
+
+internal data class CredentialProcessOutput(val accessKeyId: String, val secretAccessKey: String, val sessionToken: String?, val expiration: Instant?)
+
+internal abstract class CredentialProcessOutputParser {
+    abstract fun parse(input: String): CredentialProcessOutput
+}
+
+internal object DefaultCredentialProcessOutputParser : CredentialProcessOutputParser() {
+    private val mapper = jacksonObjectMapper()
+        .enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES)
+        .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
+        .registerModule(JavaTimeModule())
+
+    override fun parse(input: String): CredentialProcessOutput = mapper.readValue(input)
```

Comment:
Port the sanitize logic: https://github.com/aws/aws-sdk-java-v2/pull/1166/files#diff-f0e43a150bd11c291e828f1ecdb3ffd1R90